front/lib/resources/data_source_resource.ts

```diff
@@ -155,6 +155,14 @@ export class DataSourceResource extends ResourceWithVault<DataSource> {
     });
   }
 
+  static async listByVaults(auth: Authenticator, vaults: VaultResource[]) {
```

Comment:
We already have a `listByVault` can we ensure that it calls this `listByVaults` so we avoid duplicated implementation?